### PR TITLE
Add support for SQL Server connection URI

### LINKED ISSUE #797: MSSQL Missing method MSSQLConnectOptions#fromUri
All the others `*ConnectinoOptions`  have it.

Example: https://github.com/eclipse-vertx/vertx-sql-client/blob/master/vertx-pg-client/src/main/java/io/vertx/pgclient/PgConnectOptions.java#L53

### PR DESCRIPTION
**Pull request by BillyYccc**
Motivation:

Fixes #797

Conformance:

Your commits should be signed and you should have signed the Eclipse Contributor Agreement as explained in https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md
Please also make sure you adhere to the code style guidelines: https://github.com/vert-x3/wiki/wiki/Vert.x-code-style-guidelines

### CODE REVIEW

**Comment by tsegismont** [comment]
@BillyYccc @vietj what's missing before merging?

**Comment by tsegismont** [comment]
The SQL Server JDBC Client format looks simple enough.

Why block this PR until we find solutions for the limitations listed in the RFC?

**Comment by BillyYccc** [comment]
Note this PR does not provide a parser for the jdbc format but defines a
new customized format like what MySQL and postgres does.

Thomas Segismont ***@***.***> 于 2021年3月24日周三 23:38写道：

**Comment by tsegismont** [comment]
Le mer. 24 mars 2021 à 17:51, Billy Yuan ***@***.***> a
écrit :

> Note this PR does not provide a parser for the jdbc format but defines a
> new customized format like what MySQL and postgres does.
>

Right.

So IIUC, the only blocker for this PR to be merged is a code review?

**Comment by BillyYccc** [comment]
yes we can push this forward first

Thomas Segismont ***@***.***> 于 2021年3月25日周四 01:43写道：

**Comment by tsegismont** [comment]
OK, I'll make a review today then.

Le jeu. 25 mars 2021 à 02:34, Billy Yuan ***@***.***> a
écrit :

File: vertx-mssql-client/src/main/asciidoc/index.adoc
Comment (tsegismont) [inline vertx-mssql-client/src/main/asciidoc/index.adoc:106]:
```suggestion
=== Connection URI
```

File: vertx-mssql-client/src/main/asciidoc/index.adoc
Comment (tsegismont) [inline vertx-mssql-client/src/main/asciidoc/index.adoc:108]:
```suggestion
As an alternative to configuring the client with a {@link io.vertx.mssqlclient.MSSQLConnectOptions} data object, you can use a connection URI:
```

File: vertx-mssql-client/src/main/asciidoc/index.adoc
Comment (tsegismont) [inline vertx-mssql-client/src/main/asciidoc/index.adoc:115]:
```suggestion
The connection URI format is defined by the client in an idiomatic way: `sqlserver://[user[:[password]]@]host[:port][/database][?attribute1=value1&attribute2=value2...]`
```

**Review by tsegismont — APPROVED** [review]
LGTM, thank you @BillyYccc . Just a couple of suggestions.

**Comment by BillyYccc** [comment]
Thanks for the review

**Pull request merged**
